pkg/controllers/datastoresyncer/datastoresyncer.go

```diff
@@ -31,8 +33,8 @@ type DatastoreSyncer struct {
 	submarinerClusterInformer  submarinerInformers.ClusterInformer
 	submarinerEndpointInformer submarinerInformers.EndpointInformer
 	datastore                  datastore.Datastore
-	localCluster               types.SubmarinerCluster
-	localEndpoint              types.SubmarinerEndpoint
+	localCluster               *types.SubmarinerCluster
+	localEndpoint              *types.SubmarinerEndpoint
```

Comment:
Could you explain this in the commit message?

Reply:
It was to avoid more copying of the objects but it's negligible and really shouldn't be part of this PR. I changed it back.